### PR TITLE
Material holding machines can't connect to ore silos on different z levels

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

The station's ore silo can't connect to machines that are off the station z-level anymore. This doesn't affect multi-z stations.
Also includes ore silo code improvement because I was unhappy with it.

https://cdn.discordapp.com/attachments/279359975250198528/974701591137509456/2022-05-13_11-55-20.mp4

## Why It's Good For The Game

When we removed part upgrades, we also removed the main reason Miners kept the ORM on the station. There's a specific access for jobs to access the ORM - ACCESS_MINERAL_STOREROOM - but they can't take advantage of it, because Miners always bring the ORM down to Lavaland to never interact with the station, literally ever.

I tied this to the station's ore silo, because I didn't want to limit ghost roles from using ore silos for their own stuff.

## Changelog

:cl:
balance: Station equipment that holds materials (techfabs, ORMs) can't connect to ore silo's on a different Z level.
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
miners now need to take a 30 second shuttle ride whenever they want to deposit their ores so that they can continue mining :/

**Comment by JohnFulpWillard** [comment]
> miners now need to take a 30 second shuttle ride whenever they want to deposit their ores so that they can continue mining :/

ok

**Comment by SmArtKar** [comment]
imo this will lower miner efficiency even more, causing more screaming at them from the station

**Comment by JohnFulpWillard** [comment]
> imo this will lower miner efficiency even more, causing more screaming at them from the station

If we wanted mining efficiency to be maximized by having the ORM be on Lavaland, why wouldn't we just set the ORM to spawn on lavaland instead of on-station...

**Comment by the-orange-cow** [comment]
In you "Why it's good for the game" section it seems as if the only reason presented is that since there is an access to the ORM, the ORM should be available at all times.  Almost every job on the station has access to a prolathe or tech fab, which could be used to get materials already.

It seems as if the focus here is more on altering miner game flow than the stated reason of solving the nonissue of material access.

**Comment by private-tristan** [comment]
> there's a specific access for jobs to access the ORM - ACCESS_MINERAL_STOREROOM 

that access had its only use in the times before ore silo to stop assistants from going up to ORM and stealing everything out of it

**Comment by JohnFulpWillard** [comment]
> It seems as if the focus here is more on altering miner game flow than the stated reason of solving the nonissue of material access.

I made this PR because I actually use the ORM often. It is a convinient way of getting plasteel and other alloys that I otherwise can't get as the Service, Medical, and Security techfabs do not print them. Whenever I'm a non-engineer/sci/cargo tech and need alloys, the ORM is never there.
It's also a convinient way to get materials without direct cargo access, to cover up hull breaches and such on the spot, rather than leave it siphoning while you go to your department for materials. I would tell you how handy this would be in-game, but I can't because the ORM is never accessible by me.

For these two reasons, as a Shaft Miner I am constantly fighting and arguing to keep the ORM on-station, because every other Miner wants it on the mining base so they never have to go to the station on a space station game.

tl;dr, I have access to something and I want to be able to access it without having to deal with it being taken away roundstart.

**Comment by castawaynont** [comment]
you don't have access to the orm as a non cargonian, no?

**Comment by JohnFulpWillard** [comment]
> you don't have access to the orm as a non cargonian, no?

Everyone but assistants have access to it

**Comment by magatsuchi** [comment]
frankly this seems like a bad solution to a problem that does indeed exist. it should be dealt with a different way.

**Comment by castawaynont** [comment]
if you want to make miners interact with the station more, do it in an actually substative way. This does not make miners interact with the station more, just adds tedium to the job and fucks the station if cargo isn't powered.

**Comment by Shadowflame909** [comment]
Gonna pull an unusual here and agree with this nerf. Miners should have to go back on station to deposit their ores, interact with the game as a whole more.

We all know that time we had a miner as an antag target but didn't want to risk going into lavaland or definitely icebox without the gear they get. If antags always know that you have to come to this spot. It can lead to some neat trap setups and plans.

**Comment by MetalClone** [comment]
There are protolathes in every single department, and if your justification for a fairly strong inconvenience for miners (a job that regularly has no players beneath highpop hours) is that people with ORM access from service can't get alloys, it's not really that strong. There are usually 2-3 sources of plasteel on station to start the round with, and the rest of the alloys can just be obtained by asking someone in science or engineering. Also it takes literally less than 3 minutes for a quartermaster to build a new ORM and link it up to the ore silo.

**Comment by private-tristan** [comment]
> If antags always know that you have to come to this spot. It can lead to some neat trap setups and plans.

antags already know that you'll head back to the mining base (where the ORM gets put) they can just hide there instead
throw out a soap under one of the doors, hide in a locker or crate, watch you slip, and then use  stunbaton on you twice, while optionally disabling miner comms

**Comment by JohnFulpWillard** [comment]
> antags already know that you'll head back to the mining base (where the ORM gets put) they can just hide there instead throw out a soap under one of the doors, hide in a locker or crate, watch you slip, and then use stunbaton on you twice, while optionally disabling miner comms

Every single person who plays antagonist already knows they have to [insert complicated set of rules that expect you to have certain accesses and tools at your disposal ahead of time].

**Comment by Pickle-Coding** [comment]
I agree with the change, but my reason being different than the author's reason. I think it's nice to have a normal thing to do, like traveling back to station to deliver materials after mining a batch.

When things get too efficient, such as being able to instantly deliver materials between different z-levels, then it kind of detracts your experience of providing materials to the station, or doing the job.

File: code/datums/components/remote_materials.dm [hunk 1]
```diff
@@ -105,19 +106,31 @@ handles linking back and forth.
 		if (silo == M.buffer)
 			to_chat(user, span_warning("[parent] is already connected to [silo]!"))
 			return COMPONENT_BLOCK_TOOL_ATTACK
+		var/turf/user_loc = get_turf(user)
+		if(GLOB.ore_silo_default == M.buffer && !is_station_level(user_loc.z))
```
Comment (francinum) [inline code/datums/components/remote_materials.dm:110]: Since `GLOB.ore_silo_default` only takes into account the roundstart silo, and not replacements, for 3 seconds of work that mining is clearly already willing to put in (and then saddling engineering with the work of actually reconnecting the lathes), it's unlikely this will have the intended effect.

Comment (JohnFulpWillard) [reply code/datums/components/remote_materials.dm:110]: Don’t see it as a problem. Ore silo is behind a bolted QM door (and can’t be deconstructed normally) and it is a lot of time/effort  just to be able to bring an ORM down, compared to taking a wrench you spawn next to and taking a machine 2 steps away

Comment (francinum) [reply code/datums/components/remote_materials.dm:110]: > Don’t see it as a problem. Ore silo is behind a bolted QM door (and can’t be deconstructed normally) and it is a lot of time/effort just to be able to bring an ORM down, compared to taking a wrench you spawn next to and taking a machine 2 steps away

...The ore silo can be deconstructed just like any other machine? It responds to the standard deconstruction steps and procs like most other machines. And I doubt mining would mind the minor extra work of bothering the AI to open the vault.

Comment (JohnFulpWillard) [reply code/datums/components/remote_materials.dm]: Sorry it took a while, I wanted to look further into this and just never got around to it, but I have now.

You're right that it can be deconstructed, I thought otherwise from my time playing as QM against swarmers, my memory is probably just terrible.
I didn't see the Shaft Miner issue as much of a problem, because at that point it's sorta an admin-issue if Shaft Miners are breaking into high-risk areas and disconnecting ore silos just because they don't want to spend 30 seconds going to the station to drop mats off, but I changed it anyways because I don't personally think that ghost roles would have machines on separate levels anyways.

**Comment by Ryusenshu** [comment]
> A good miner is a miner that's bringing in ores quickly also there is 0 interaction still nothing has changed, i assume you mean "player interaction" when you say interact with the station. but all this does is inconvenience miners, all they did before was go to lavaland spend like 10mins or so mining then come back to cargo and spend 30 seconds printing gear from the lathe (30 seconds if you add the shuttle trip). now they probably spend like a extra 15 seconds in cargo because the ORM cant be on lavaland anymore.
> 
> Also what about space bases? there are people who may make bases in space and they may even put a lathe or a exosuit fab in there but now they can't because of this PR which targets miners

funnily enough this targets anyone but miners
miners still get their points and they have no need to care about the station mats after the first uranium sheet

shit pr

**Comment by antropod** [comment]
another non problem that was fixed
just give them second ORM

**Comment by Dmeto** [comment]
Should have made the machine unwrenchable
